docs/architecture/017-applicant-ux.md

```diff
@@ -0,0 +1,89 @@
+# 017: Applicant Form UX
+
+Authors: @opqdonut, @Macroz, @marharyta
+
+This ADR aims to collect together some topics like validations in
+frontend, autosaving, etc. that are related to the Applicant's user
+experience. There have been previous discussions about all of these
+topics but no unified presentation.
```

Comment:
this document is missing some information. what kind of validations do we need? currently fields are either required or optional, which is enforced by API, and some fields have length limitation, which seems to be a hard limit.

second question, who attaches the validations to fields? would this be the owner who creates the form?

Reply:
There is small stuff like IP address and phone number format currently. All of them are handled in API and some would be nice to see already earlier, perhaps in data entry or so.

There is no attachment step as you mean. You can set field type or set the required, max length etc. and validation will be added.